sync_ci/pkg/detect/detect.go

```diff
@@ -0,0 +1,363 @@
+				recentCaseSet[repo] = map[string][]string{}
+			}
+			if matched, name := parser.MatchAndParseSQLStmtTest(c); matched {
+				recentCaseSet[repo][name] = caseSet[repo][c]
+			} else {
+				recentCaseSet[repo][c] = caseSet[repo][c]
+			}
+		}
+	}
+}
+
+func getHistoryCases(rows *sql.Rows, caseSet map[string]map[string][]string, baselink string) {
+	repoPrCases := map[string]map[string]string{} // repo -> pr -> case
+	for rows.Next() {
+		var rawCase []byte
+		var cases []string
+		var pr string
+		var repo string
+		var jobid string
+		var job string
+		_ = rows.Scan(&repo, &pr, &rawCase, &jobid, &job)
+		if pr == "0" {
+
+		}
+		_ = json.Unmarshal(rawCase, &cases)
+		for _, c := range cases {
+			if _, ok := repoPrCases[repo]; !ok {
+				repoPrCases[repo] = map[string]string{}
+			}
+			if _, ok := repoPrCases[repo][c]; !ok {
+				repoPrCases[repo][c] = pr
+			} else {
+				if _, ok = caseSet[repo]; !ok {
+					caseSet[repo] = map[string][]string{}
+				}
+				caseSet[repo][c] = append(caseSet[repo][c], fmt.Sprintf(baselink, job, jobid))
+			}
+		}
+	}
+}
+
+func MentionIssue(cfg model.Config, repo string, issueId string, joblink string, test bool) error {
+	req := requests.Requests()
+	req.SetTimeout(10 * time.Second)
+	req.Header.Set("Authorization", "token "+cfg.GithubToken)
+	var url string
+	if !test {
+		url = fmt.Sprintf("https://api.github.com/repos/%s/issues/%s/comments", repo, issueId)
+	} else {
+		url = "https://api.github.com/repos/kivenchen/klego/issues/1/comments"
+	}
+
+	for i := 0; i < 3; i++ {
+		println("Posting to ", url)
+		resp, err := req.PostJson(url, map[string]string{
+			"body": "#" + issueId + ": new failure !(jenkins link)[" + joblink + "]", // todo: fill content templates
+		})
+		if err != nil {
+			return err
+		} else {
+			if resp.R.StatusCode != 201 {
+				return fmt.Errorf("create comment failed")
+			} else {
+				log.S().Info("Created comment %s/#%s mentioning %s", repo, issueId, joblink)
+				break
+			}
+		}
+	}
+
+	return nil
+}
+
+func CreateIssueForCases(cfg model.Config, issues []*model.CaseIssue, test bool) error {
+	req := requests.Requests()
+	req.SetTimeout(10 * time.Second)
+	req.Header.Set("Authorization", "token "+cfg.GithubToken)
+	dbIssueCase, err := SetupDB(cfg.CaseDsn)
+	if err != nil {
+		return err
+	}
+	// todo: set request header
+	for _, issue := range issues {
+		var url string
+		if !test {
+			url = fmt.Sprintf("https://api.github.com/repos/%s/issues", issue.Repo)
+		} else {
+			url = "https://api.github.com/repos/kivenchen/klego/issues"
+		}
+		var resp *requests.Response
+		for i := 0; i < 3; i++ {
+			println("Posting to ", url)
+			resp, err = req.PostJson(url, map[string]string{
+				"title":  issue.Case.String + " failed",
+				"body":   "Latest build: !(Jenkins)[" + issue.JobLink.String + "]", // todo: fill content templates
+				"labels": "component/test",
+			})
+			if err != nil {
+				return err
+			} else {
+				if resp.R.StatusCode != 201 {
```

Comment:
may be check the 2xx status code

Reply:
The GitHub API doc ensures the "201 created" success return.